enumerable.rb

```diff
@@ -0,0 +1,89 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    size.times { |n| yield (self[n]) }
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    size.times { |n| yield self[n], n }
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    result_array = []
+    my_each { |n| result_array << n if yield(n) }
+    result_array
+  end
+
+  def my_all
+    return true unless block_given?
+
+    my_each { |n| return false unless yield(n) }
+    true
+  end
+
+  def my_any
+    return true unless block_given?
```

Comment:
### **my_any**

- [ ] It should return true if the block ever returns a value other than false or nil

- [ ] It should not mutate the original array

- [ ] When no block or argument is given it should return true if at least one of the collection is not false or nil

- [ ] When a class is passed as an argument it should return true if at least one of the collection is a member of such class

- [ ] When a Regex is passed as an argument it should return false if none of the collection matches the Regex

- [ ] When a pattern other than Regex or a Class is given it should return false if none of the collection matches the pattern

Reply:
all ruby doc tests passed 
![passed2](https://user-images.githubusercontent.com/51087315/83947883-62e8cd80-a822-11ea-9270-c459d431fb07.png)